File: ios/MenuMaker/Data/Models/BusinessModels.swift
import Foundation

// MARK: - Business Models

struct Business: Codable, Identifiable {
    let id: String
    let name: String
    let slug: String
    let description: String?
    let logoUrl: String?
    let ownerId: String
    let isActive: Bool
    let createdAt: String
    let updatedAt: String

    var formattedCreatedDate: String {
        guard let date = ISO8601DateFormatter().date(from: createdAt) else {
            return createdAt
        }

        let formatter = DateFormatter()
        formatter.dateStyle = .medium
        return formatter.string(from: date)
    }

    var displayDescription: String {
        description ?? "No description available"
    }
}

struct BusinessResponse: Decodable {
    let success: Bool
    let data: BusinessData
}

struct BusinessData: Decodable {
    let business: Business
}

struct BusinessListResponse: Decodable {
    let success: Bool
    let data: BusinessListData
}

struct BusinessListData: Decodable {
    let businesses: [Business]
}

struct CreateBusinessRequest: Encodable {
    let name: String
    let slug: String
    let description: String?
    let logoUrl: String?
}

struct UpdateBusinessRequest: Encodable {
    let name: String?
    let description: String?
    let logoUrl: String?
    let isActive: Bool?
}

File: android/app/src/main/kotlin/com/menumaker/data/remote/models/BusinessModels.kt
package com.menumaker.data.remote.models

import com.google.gson.annotations.SerializedName

data class BusinessDto(
    @SerializedName("id") val id: String,
    @SerializedName("name") val name: String,
    @SerializedName("slug") val slug: String,
    @SerializedName("description") val description: String?,
    @SerializedName("logo_url") val logoUrl: String?,
    @SerializedName("owner_id") val ownerId: String,
    @SerializedName("is_active") val isActive: Boolean,
    @SerializedName("created_at") val createdAt: String,
    @SerializedName("updated_at") val updatedAt: String
)

data class BusinessResponse(
    @SerializedName("success") val success: Boolean,
    @SerializedName("data") val data: BusinessData
)

data class BusinessData(
    @SerializedName("business") val business: BusinessDto
)

data class BusinessListResponse(
    @SerializedName("success") val success: Boolean,
    @SerializedName("data") val data: BusinessListData
)

data class BusinessListData(
    @SerializedName("businesses") val businesses: List<BusinessDto>
)
